### ISSUE TITLE
False positive for sklearn vectorizer constructor parameters

### ISSUE THREAD

**Issue by edricgarran** [microsoft/pyright#1407]
**Describe the bug**
Pyright gives the "no parameter named" error for sklearn text vectorizers, which have the parameters clearly defined in their constructors, but use a decorator for positional parameter deprecation.

**To Reproduce**
```python
from sklearn.feature_extraction.text import TfidfVectorizer
TfidfVectorizer(max_df=0.8)
                [Pyright reportGeneralTypeIssues] [E] No parameter named "max_df"
```

**Reference constructor code**
```python
    @_deprecate_positional_args
    def __init__(self, *, input='content', encoding='utf-8',
                 decode_error='strict', strip_accents=None, lowercase=True,
                 preprocessor=None, tokenizer=None, analyzer='word',
                 stop_words=None, token_pattern=r"(?u)\b\w\w+\b",
                 ngram_range=(1, 1), max_df=1.0, min_df=1,
                 max_features=None, vocabulary=None, binary=False,
                 dtype=np.float64, norm='l2', use_idf=True, smooth_idf=True,
                 sublinear_tf=False):
```

**Expected behavior**
Pyright should either ignore the decorator and use the actual parameters accepted by the method as a reference, or skip typechecking entirely since the constructor isn't type-hinted anyways.

**VS Code extension or command-line**
Running through coc-pyright on neovim.

**Comment by edricgarran** [comment]
Nevermind, investigating further, it was coc-pyright overriding the default value for `useLibraryCodeForTypes`, which should be false.
